ansible_collections/arista/avd/roles/eos_designs/doc/custom-structured-configuration.md

```diff
@@ -191,3 +190,157 @@ name_server:
   - 10.10.10.11
   - 10.10.10.12
 ```
+## Variables Precedence
+
+_**TL;DR:** eos_cli_config_gen variables in the input files will be overwritten
+by the eos_designs generated facts when the role produces a value for the same
+key, use custom_strutcured_configuration to make sure the values make it in the intended configurations._
+
+### Overview
+
+AVD `eos_designs` role flow is expected to work as follow:
+
+1. from the input variables (inventory, group_vars, host_vars,...), a task first sets ansible facts for each device
+2. the facts are then used in a subsequent task to generate a structured_config
+3. this structured_config is what is then used by `eos_cli_config_gen` role to generate the intended config
+
+A caveat is that it is possible to define in the inventory, group_vars, host_vars,... some variables that are
+used by `eos_designs` and others used by `eos_cli_config_gen` roles. And while it is perfectly acceptable,
+the behavior can sometimes be confusing. If the key name for an `eos_cli_config_gen` variable in the variable files is the same as one of
+the key name of a variable generated by `eos_designs` steps, only the value from `eos_designs` step will be configured.
+However if there is no clash, then the value of the variable will be the one defined in the variable files.
+
+The consequence is that upgrading AVD to a newer version for instance, or configuring additional `eos_designs` functionnality in some existing AVD repo,
+could lead to some additional variable keys being generated by `eos_designs` in the structured_config, overwriting previously working `eos_cli_config_gen` keys.
+
+<!-- ![Figure 1: Variables Precedence between roles](../../../media/variables_precedence.png) -->
+
+<div style="text-align:center">
+  <img src="../../../media/variables_precedence.png" />
+</div>
+
+The recommendation is to use custom_structured_configuration as described in [the following section of this guide](./custom-structured-configuration.md). This configuration will always prevails over the `eos_designs` generated variables.
```

Comment:
After you move this up the vars for custom_structured_configuration will be below on this page.
```suggestion
The recommendation is to use `custom_structured_configuration` as described below. This configuration will be merged on top of the `eos_designs` generated variables.
```